pkg/jobworker/cgroup.go

```diff
@@ -0,0 +1,62 @@
+package jobworker
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"syscall"
+)
+
+// Cgroup implements ResourceController and provides a minimal interface for the host's cgroup
+type Cgroup struct {
+	rootPath string
+}
+
+// AddProcess mutates the given cmd to instruct go to add the PID of the started process to a given cgroup
+func (cg *Cgroup) AddProcess(name string, cmd *exec.Cmd) error {
+	// Add job's process to cgroup
+	f, err := syscall.Open(cg.groupPath(name), 0x200000, 0)
```

Comment:
Yes that's correct, we have 1 cgroup per running job. I also thought of this, but it would involve having 1 cgroup per JobOpts, where the CPU, memory and IO would all have to be the same value across jobs.

We could instead have a map of JobOpts key'd by cgroup ID, so before creating a cgroup we could check if one exists that would match the resource controls. This does make it harder though to update the resource controls as jobs would need to be shuffled across cgroups if some jobs needed more resource after running.